lab-nassir/README.md

```diff
@@ -0,0 +1,44 @@
+![cf](https://i.imgur.com/7v5ASc8.png) Javascript 401d10 -- Lab 13-14
+======
+
+## Express API -- CatAPI 3.0 'Saruman'
+
+CatAPI announces CatAPI 3.0 'Sauruman'! We're not only leapfrogging our competition, we're entering a NEW DIMENSION of API-based cat-action -- or 'Catction' for short.
+
+CatAPI is built using Express, and is tested using SuperAgent.
+
+# What's new in 3.0 'Saruman'
+
+CatAPI will now ape (or 'cat-ape') prevailing Silicon Valley conventions by assigning cutesy, themed codenames to our versions.
+
+CatAPI 3.0 'Saruman' now tracks cat cafes! You will no longer be able to track your own personal cats. Each cat recorded by CatAPI 3.0 must be associated with a cat cafe. All your existing cat records are now associated with a cat cafe, and the cat cafe authorities will arrive soon to collect them for transportation to their new home.
+
+# Install the API
+
+From the home directory, run `npm i` in terminal. All required dependencies will install.
+
+# Start the API
+
+From the home directory, run `npm start` in terminal. A welcome message will be sent and the API is ready to use.
+
+# Test the API
+
+From the home directory, run `npm test` in terminal. A test using Mocha, Chai and Superagent will run. If any tests fail, see `What to do if you encounter a bug` below in this readme.
+
+# API endpoints
+
+CatAPI has two endpoints:
+
+`/api/cafe/:cafeId` for cat-cafe requests. Accepts GET, POST, PUT and DELETE requests. Calling this endpoint with no cafeId returns a list of existing cafe IDs.
```

Comment:
You should specify what a client needs to send with a POST request in terms of properties and the data types associated with their values (same for cats).